react/src/lib/components/SubsurfaceViewer/layers/axes2d/axes2DLayer.ts

```diff
@@ -154,8 +167,209 @@ export default class Axes2DLayer extends Layer<Axes2DLayerProps<unknown>> {
+
+        // Add minor X ticks.
+        if (ticks.length > 1) {
+            const tick1 = ticks[0];
+            const tick2 = ticks[1];
+            const d = (tick2 - tick1) / (n_minor_ticks + 1);
+            const tick_start = tick1;
+
+            // up
+            let i = 0;
+            while (tick_start + (i + 1) * d < max) {
+                const tick = tick_start + (i + 1) * d;
+                tick_labels.push("");
+                i++;
+
+                if (isHorizontal) {
+                    lines.push(tick, y_tick, z_tick); // tick line start
+                    lines.push(tick, y_tick + 0.5 * tick_length, z_tick); // tick line end.
+                } else {
+                    lines.push(x_tick, tick, z_tick);
+                    lines.push(x_tick + 0.5 * tick_length, tick, z_tick);
+                }
+            }
+
+            // down
+            i = 0;
+            while (tick_start - (i + 1) * d > min) {
+                const tick = tick_start - (i + 1) * d;
+                tick_labels.push("");
+                i++;
+
+                if (isHorizontal) {
+                    lines.push(tick, y_tick, z_tick);
+                    lines.push(tick, y_tick + 0.5 * tick_length, z_tick);
+                } else {
+                    lines.push(x_tick, tick, z_tick);
+                    lines.push(x_tick + 0.5 * tick_length, tick, z_tick);
+                }
+            }
+        }
+
+        const labels = this.makeLabelsData(lines, tick_labels);
+
+        return [lines, labels];
+    }
+
+    GetBacgroundTriangleLinesHorizontal(
+        x_min_w: number,
+        x_max_w: number,
+        isTop: boolean,
+        viewMatrix: number[],
+        pixel2world: number
+    ): number[] {
+        const mv = this.props.marginV * pixel2world;
+
+        const vp_bounds = this.context.viewport.getBounds(); // [xmin, ymin, xmax, ymax]
+
+        const y_max = isTop ? vp_bounds[3] : vp_bounds[1] + mv;
+        const y_min = isTop ? vp_bounds[3] - mv : vp_bounds[1];
+
+        const z = 0;
+        const p1 = vec4.fromValues(x_min_w, y_max, z, 1);
+        const p2 = vec4.fromValues(x_max_w, y_max, z, 1);
+        const p3 = vec4.fromValues(x_max_w, y_min, z, 1);
+        const p4 = vec4.fromValues(x_min_w, y_min, z, 1);
+
+        const p1_v = word2view(viewMatrix, p1);
+        const p2_v = word2view(viewMatrix, p2);
+        const p3_v = word2view(viewMatrix, p3);
+        const p4_v = word2view(viewMatrix, p4);
+
+        // Distance camera background in view space.
+        const z_dist = 101;
+        p1_v[2] = -z_dist;
+        p2_v[2] = -z_dist;
+        p3_v[2] = -z_dist;
+        p4_v[2] = -z_dist;
+
+        /*eslint-disable */
+        const background_lines: number[] = [ 
+            ...p1_v, ...p2_v, ...p4_v,  // triangle 1
+            ...p2_v, ...p4_v, ...p3_v,  // triangle 2 
+        ];
+        /*eslint-enable */
+
+        return background_lines;
+    }
+
+    GetBacgroundTriangleLinesVertical(
```

Comment:
This is a lot of repeated code from `GetBacgroundTriangleLinesHorizontal`. Could we generalize more?